corehq/apps/hqwebapp/static/hqwebapp/js/ckeditor_knockout_bindings.js

```diff
@@ -0,0 +1,185 @@
+                        CKEditor5.image.ImageStyle,
+                        CKEditor5.image.ImageResize,
+                        CKEditor5.image.ImageResizeButtons,
+                        CKEditor5.image.ImageToolbar,
+                        CKEditor5.image.ImageUpload,
+                        CKEditor5.indent.Indent,
+                        CKEditor5.link.Link,
+                        CKEditor5.link.LinkImage,
+                        CKEditor5.list.List,
+                        CKEditor5.paragraph.Paragraph,
+                        CKEditor5.pasteFromOffice.PasteFromOffice,
+                        CKEditor5.restrictedEditing.RestrictedEditingMode,
+                        CKEditor5.upload.SimpleUploadAdapter,
+                    ],
+                    toolbar: {
+                        items: [
+                            'heading',
+                            'fontFamily',
+                            'fontSize',
+                            'fontColor',
+                            '|',
+                            'bold',
+                            'italic',
+                            'link',
+                            'alignment',
+                            'bulletedList',
+                            'numberedList',
+                            'uploadImage',
+                            '|',
+                            'outdent',
+                            'indent',
+                            '|',
+                            'undo',
+                            'redo',
+                            'restrictedEditing',
+                        ],
+                    },
+                    image: {
+                        insert: {
+                            type: 'inline',
+                        },
+                        toolbar: [
+                            'imageStyle:side',
+                            '|',
+                            'toggleImageCaption',
+                            '|',
+                            'linkImage',
+                        ],
+                    },
+                    simpleUpload: {
+                        uploadUrl: initialPageData.reverse(element.attributes['data-image-upload-url'].value),
+                        withCredentials: true,
+                        headers: {
+                            'X-CSRFTOKEN': $("#csrfTokenContainer").val(),
+                        },
+                    },
+                    htmlSupport: {
+                    // We allow all HTML here, and filter it out in a sanitizing step
+                        allow: [
+                            {
+                                name: /.*/,
+                                attributes: true,
+                                classes: true,
+                                styles: true,
+                            },
+                        ],
+                    },
+                    restrictedEditing: {
+                        allowedCommands: [
```

Comment:
out of curiosity, what kinds of commands are restricted?

Reply:
I didn't explicitly leave anything out of the list. In restricted editing mode, you have to allowlist all the functionality you want a user to do. I went through the list of "plugins" we are using and added each command they define here.